src/main/java/net/dv8tion/jda/api/interactions/components/text/Modal.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.interactions.components.text;
+
+import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
+import net.dv8tion.jda.api.interactions.ModalInteraction;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.ItemComponent;
+import net.dv8tion.jda.api.utils.data.SerializableData;
+import net.dv8tion.jda.internal.interactions.component.ModalImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nonnull;
+import java.util.*;
+
+/**
+ * Represents a Discord Modal
+ *
+ * <p>Replying to an interaction with a modal will cause a form window to pop up on the User's Discord client.
+  *
+  * <h2>Example</h2>
+ * <pre>{@code
+ *     public void onSlashCommandInteraction(@NonNull SlashCommandInteractionEvent event)
+ *     {
+ *         if (event.getName().equals("support"))
+ *         {
+ *             TextInput email = TextInput.create("email", "Email", TextInputStyle.SHORT)
```

Comment:
Maybe we can find a better example that doesn't suggest grabbing private information of users